### PR TITLE
[docs] PEFT + Transformer docs

### PR DESCRIPTION
**Pull request by stevhliu**
Thanks for kickstarting the docs @younesbelkada! 

I've iterated a bit on it and added the `from_pretrained` example and a link to the main 🤗 PEFT docs for users who're interested to learn more. Depending on how close/ready this [PR](https://github.com/huggingface/transformers/pull/24750) is for also loading PEFT adapters from the pipeline, I think it may be nice to also include a section here about that (but obviously no rush 😄 ).

### CODE REVIEW

File: docs/source/en/peft.md [hunk 1]
```diff
@@ -42,6 +40,26 @@ If you want to try out the brand new features, you might be interested in instal
...
+```py
+from transformers import AutoModelForCausalLM, AutoTokenizer
+
+peft_model_id = "ybelkada/opt-350m-lora"
+model = AutoModelForCausalLM.from_pretrained(peft_model_id)
+```
+
+<Tip>
+
+You can still load a PEFT adapter even if your model isn't an auto-mapped model as long as the Hub repository or your local directory contains an `adapter_config.json` file with the `base_model_name_or_path` key.
```
Comment (stevhliu) [inline docs/source/en/peft.md:59]: Feel free to provide more context for why a model wouldn't be auto-mapped already!

File: docs/source/en/peft.md [hunk 2]
```diff
@@ -42,6 +40,26 @@ If you want to try out the brand new features, you might be interested in instal
...
+
+</Tip>
+
 <!--
 
 TODO: (@younesbelkada @stevhliu)
```
Comment (stevhliu) [inline docs/source/en/peft.md:65]: For the `Trainer` integration, did you mean something like:

```py
model_name = "roberta-large"
peft_config = PromptEncoderConfig(task_type="SEQ_CLS", num_virtual_tokens=20, encoder_hidden_size=128)
model = AutoModelForSequenceClassification.from_pretrained(model_name)
model = get_peft_model(model, peft_config)
trainer = Trainer(model=model, ...)
trainer.train()
```

Comment (younesbelkada) [reply docs/source/en/peft.md:65]: Exactly yes ! Although it seems most of the usage seems to be around LoRA models , but overall yes if you pass a `PeftModel` to trainer it should just work like that

**Review by younesbelkada — APPROVED** [review]
Thanks so much for iterating on the docs!

**Pull request merged**